jujugui/static/gui/src/app/views/utils.js

```diff
@@ -1822,31 +1846,40 @@ YUI.add('juju-view-utils', function(Y) {
   */
   utils.generateServiceName = function(
     charmName, charmId, services, ignoreSelf) {
-    var charmCount = 0;
     // Loop through each service and grab its charmid to see if it matches.
-    services.each((service) => {
+    var highestCounter = Math.max.apply(Math, services.map((service) => {
```

Comment:
Can you expand the reasoning behind this approach in a comment. Something like... "We can't simply count the number of charms deployed because a user may remove a service in the middle of the charms and this causes a duplication of the 'next' id"